projects/power_distribution/test/test_front_uv_detector.c

```diff
@@ -0,0 +1,55 @@
+#include "can_transmit.h"
+#include "front_uv_detector.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "ms_test_helpers.h"
+#include "pd_events.h"
+
+static GpioAddress s_uv_comp_pin_address = { .port = GPIO_PORT_B, .pin = 0 };
+
+static volatile bool s_interrupt_ran = false;
+
+static CanStorage s_can_storage;
+
+static void prv_initialize_can(SystemCanDevice can_device) {
+  CanSettings can_settings = {
+    .device_id = can_device,
+    .loopback = true,
+    .bitrate = CAN_HW_BITRATE_500KBPS,
+    .rx_event = POWER_DISTRIBUTION_CAN_EVENT_RX,
+    .tx_event = POWER_DISTRIBUTION_CAN_EVENT_TX,
+    .fault_event = POWER_DISTRIBUTION_CAN_EVENT_FAULT,
+    .tx = { GPIO_PORT_A, 12 },
+    .rx = { GPIO_PORT_A, 11 },
+  };
+  can_init(&s_can_storage, &can_settings);
+}
+
+static StatusCode prv_rx_callback(const CanMessage *msg, void *context, CanAckStatus *ack_reply) {
+  s_interrupt_ran = true;
+  return STATUS_CODE_OK;
+}
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  soft_timer_init();
+  event_queue_init();
+  front_uv_detector_init();
```

Comment:
Do a `TEST_ASSERT_OK` on this (macro from test_helpers.h) to verify that it returns OK. (Asserts are valid in setup_test and teardown_test by the way.)